lib/src/Base/Stat/SampleImplementation.cxx

```diff
@@ -2165,6 +2166,23 @@ SampleImplementation SampleImplementation::operator - (const SampleImplementatio
   return sample;
 }
 
+SampleImplementation SampleImplementation::shuffledCopy() const
+{
+  /* Shuffle an array a of n elements (indices 0..n-1) 
+    see https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm 
+  */
+  SampleImplementation sampleOut = *this ;
+  Point ptmp(sampleOut.getDimension());
+  for(UnsignedInteger i=sampleOut.getSize()-1; i>0; --i)
+  {
+    UnsignedInteger j = RandomGenerator::IntegerGenerate(i+1);
+    ptmp = sampleOut[i];
+    sampleOut[i] = sampleOut[j];
+    sampleOut[j] = ptmp;
+  }
+  return sampleOut;
+}
+
```

Comment:
Couldn't these 10 lines of code be moved to a static method of the `RandomGenerator` class? Its prototype would be something like
``static Sample shuffle(const Sample sample);``
This would make it equivalent to `numpy.random.shuffle`

Reply:
I suggest this method could stay in the HSICEstimatorImplementation for now

Reply:
As a static method?

Reply:
I implemented @jschueller suggestion. It saves a lot of discussion.